tabpy-server/server_tests/test_endpoint_handler.py

```diff
@@ -0,0 +1,111 @@
+import base64
+import hashlib
+import os
+import tempfile
+import unittest
+
+from argparse import Namespace
+from tabpy_server.app.app import TabPyApp
+from tabpy_server.handlers.endpoint_handler import EndpointHandler
+from tornado.testing import AsyncHTTPTestCase
+from unittest.mock import patch
+
+
+class TestEndpointHandlerWithAuth(AsyncHTTPTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.patcher = patch(
+            'tabpy_server.app.TabPyApp._parse_cli_arguments',
+            return_value=Namespace(
+                config=None))
+        cls.patcher.start()
+
+        prefix = '__TestEndpointHandlerWithAuth_'
+        # create password file
+        cls.pwd_file = tempfile.NamedTemporaryFile(
+            mode='w+t', prefix=prefix, suffix='.txt', delete=False)
+        cls.pwd_file.write('username {}'.format(
+            hashlib.sha3_256('password'.encode('utf-8')).hexdigest()))
+        cls.pwd_file.close()
+
+        # create state.ini dir and file
+        cls.state_dir = tempfile.mkdtemp(prefix=prefix)
+        cls.state_file = open(os.path.join(cls.state_dir, 'state.ini'), 'w+')
+        cls.state_file.write('[Service Info]\n'

```

Comment:
We do this in multiple tests. Maybe we should extract it into utils, especially as we continue to add tests for endpoint handlers.

Reply:
agree, do it with your next change